src/BatteryStateOfChargeService.h

```diff
@@ -45,5 +45,12 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
     // Convention is to have an _ after private members
     const double initialStateOfChargePercent_;
 
+    double AmpHoursUsed_;
+    double inCurrent_;
+    double newCurrent_=0;
+    double sumCurrent_=0;
+    bool firstRun_=true;
+    QTime initialTime_;
```

Comment:
Same comments as with `inCurrent_`. In particular though, `initial` is not very accurate since you are changing it every time you `addData` so it is only the "initial time" the first time `addData` is called.